### PR TITLE
fix: Data too long for column 'resource_id'

### PR DESCRIPTION
**Pull request by shadinaif**
when the course id is a little bit long, the lti xblock id becomes too long for resource_id to handle

To reproduce:
* Create a course with long ID. For example: `course-v1:Public+TEST_IT_01+TEST_IT_01_Dec_2023`
* Add `lti_consumer` in Advanced settings for the course
* Try to add an LTI component in the course with version `1.3`
* The unit crashes
Screenshot from 2023-12-25 14-00-00

we tried that on the latest https://sandbox.openedx.org

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @shadinaif! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by shadinaif** [comment]
For your review @OmarIthawi, please

**Comment by OmarIthawi** [comment]
Thanks @shadinaif please update the following:

 - version: https://github.com/openedx/xblock-lti-consumer/blob/master/lti_consumer/__init__.py
 - changelog: https://github.com/openedx/xblock-lti-consumer/blob/master/CHANGELOG.rst

**Comment by shadinaif** [comment]
please check now @OmarIthawi

File: lti_consumer/migrations/0018_auto_20231225_0556.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 3.2.23 on 2023-12-25 10:56
```
Comment (OmarIthawi) [inline lti_consumer/migrations/0018_auto_20231225_0556.py:1]: @shadinaif would you mind renaming on this migration file to something more readable similar to [other migrations](https://github.com/openedx/xblock-lti-consumer/tree/454346fb3ea0c3888b4dbb269594f2412d70a127/lti_consumer/migrations)?

**Review by OmarIthawi — COMMENTED** [review]
Thanks Shadi. One more thing please.

**Review by OmarIthawi — APPROVED** [review]
Looks good to me. Thanks @shadinaif!!

**Comment by mphilbrick211** [comment]
Hi @OmarIthawi and @shadinaif! Is this pull request still needed?

**Comment by OmarIthawi** [comment]
Yes @mphilbrick211, we needed it to get merged but it wasn't triaged since [Jan 25th](https://github.com/openedx/xblock-lti-consumer/pull/435#pullrequestreview-1843882634).

@shadinaif there's a new changelog update that conflicts with this pull request, kindly rebase and fix it.

**Comment by shadinaif** [comment]
Rebased. But codecov is failing for no reason ¯\_(ツ)_/¯

**Comment by mphilbrick211** [comment]
Hi @shadinaif and @OmarIthawi! Can this be closed?

**Comment by OmarIthawi** [comment]
Thanks @shadinaif. @mphilbrick211, this is now ready for merging. I don't have permissions though. Could you please help?

**Comment by itsjeyd** [comment]
@shadinaif @OmarIthawi Sorry for the delay here. Is this PR still needed?

@shadinaif If so, could you please resolve merge conflicts?

Once the build is green we can have someone from Axim engineering merge the PR.

**Comment by OmarIthawi** [comment]
yes it's still needed @itsjeyd

**Comment by Faraz32123** [comment]
@shadinaif a soft reminder to resolve the merge conflicts. So that it can be merged, Thanks for bearing it for a long time.

**Comment by OmarIthawi** [comment]
@Faraz32123 would you mind cherry-picking this commit and including it in the repo?
We've rebased it earlier twice as far as I know got stale and had conflicts.

**Comment by Faraz32123** [comment]
> @Faraz32123 would you mind cherry-picking this commit and including it in the repo? We've rebased it earlier twice as far as I know got stale and had conflicts.

yes, I was also thinking the same. Let's wait for @shadinaif for a day or two. Otherwise, I will cherry-pick it.

**Comment by Faraz32123** [comment]
closing this PR in favor of this PR: https://github.com/openedx/xblock-lti-consumer/pull/523

**Pull request closed without merging**